app/helpers/activities_helper.rb

```diff
@@ -63,56 +63,39 @@ def steps_without_operations_data_for_steps(steps)
   end
 
   def operations_data(operations)
-    occured_predicates = []
-    operations.reduce([]) do |memo, fact|
-      if occured_predicates.include?(fact.predicate)
-        obj = memo.select { |f| f["predicate"] == fact.predicate }.first
-        obj["repeats"] = obj["repeats"] ? obj["repeats"]+1 : 0
-        next memo
-      end
+    operations.map do |fact|
       elem = fact.object_asset
-      if elem
-        obj = { "object_asset" => {
-                 uuid: elem.uuid,
-                 barcode: elem.barcode,
-                 id: elem.id,
-                 info_line: elem.info_line
-               } }.merge(fact.attributes)
-      else
-        obj = fact.attributes
-      end
-      obj[:asset]=fact.asset.attributes
-      memo.push(obj)
-      memo
+      obj = if elem
+              { "object_asset" => {
+                uuid: elem.uuid,
+                barcode: elem.barcode,
+                id: elem.id,
+                info_line: elem.info_line
+              } }.merge(fact.attributes)
+            else
+              fact.attributes
+            end
+      obj[:asset] = fact.asset.attributes
+      obj
     end
```

Comment:
The top section of this `if occured_predicates.include?(fact.predicate)` will never be true, so this can be simplified drastically.